yt_dlp/downloader/external.py

```diff
@@ -434,21 +423,30 @@ def _call_downloader(self, tmpfilename, info_dict):
 
         start_time, end_time = info_dict.get('section_start') or 0, info_dict.get('section_end')
 
-        for i, url in enumerate(urls):
-            if http_headers is not None and re.match(r'^https?://', url):
-                args += http_headers
+        selected_formats = info_dict.get('requested_formats') or [info_dict]
+        for i, fmt in enumerate(selected_formats):
+            if fmt.get('http_headers') is not None:
+                headers_dict = handle_youtubedl_headers(fmt['http_headers'])
+                # Trailing \r\n after each HTTP header is important to prevent warning from ffmpeg/avconv:
+                # [http @ 00000000003d2fa0] No trailing CRLF found in HTTP header.
+                http_headers = [
+                    '-headers', ''.join(f'{key}: {val}\r\n' for key, val in headers_dict.items())
+                ]
+                if re.match(r'^https?://', fmt['url']):
+                    args += http_headers
```

Comment:
```suggestion
            if fmt.get('http_headers') and re.match(r'^https?://', fmt['url']):
                headers_dict = handle_youtubedl_headers(fmt['http_headers'])
                # Trailing \r\n after each HTTP header is important to prevent warning from ffmpeg/avconv:
                # [http @ 00000000003d2fa0] No trailing CRLF found in HTTP header.
                args.extend('-headers', ''.join(f'{key}: {val}\r\n' for key, val in headers_dict.items()))
```

(my indents may be wrong)